app/src/main/java/org/sopt/cgv/feature/time/component/SelectableTheatersInModal.kt

```diff
@@ -0,0 +1,206 @@
+package org.sopt.cgv.feature.time.component
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawBehind
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import kotlinx.collections.immutable.PersistentList
+import kotlinx.collections.immutable.persistentListOf
+import org.sopt.cgv.R
+import org.sopt.cgv.core.common.extension.noRippleClickable
+import org.sopt.cgv.core.designsystem.theme.CGVTheme
+import org.sopt.cgv.core.designsystem.theme.Gray200
+import org.sopt.cgv.core.designsystem.theme.Gray700
+import org.sopt.cgv.core.designsystem.theme.Gray850
+import org.sopt.cgv.core.designsystem.theme.PrimaryRed400
+import org.sopt.cgv.core.designsystem.theme.White
+
+
+data class MovieTheatersByDetailRegion(
+    val detailRegionName: String,
+    val theaterNames: PersistentList<String>
+)
+
+@Composable
+fun SelectableTheatersInModal(
+    movieTheatersByDetailRegion: PersistentList<MovieTheatersByDetailRegion>,
+    selectedTheaters: MutableState<Set<String>>,
```

Comment:
```suggestion
    selectedTheaters: Set<String>,
    onTheaterSelected: (String) -> Unit
```

여기 부분도 이렇게 수정해주시면 상태 관리 책임이 컴포넌트 외부로 이동할 수 있을 거 같아요. 지금 대부분의 코드에서 MutableState를 직접 전달하는 대신 불변 데이터와 콜백 함수를 전달한다고 보시면 쉬울 거 같습니다!